orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -830,10 +832,39 @@ class SqlExecutionRepository(
         stages.forEach { storeStageInternal(ctx, it, executionId) }
       }
     } finally {
+      // Restore original object state.
```

Comment:
the fact that we need to do this seems a bit wild to me - I wonder if it's worth just objectMapper copying the execution on the way in here so we are free to mutate it as we see fit..

unless the expectation is that the repository mutates to execution to set Ids on it or something...

anyway not super relevate to the overall change in this PR

Reply:
This was an optimization. Saving the execution is a high-throughput (as far as Orca is concerned) operation, and serialization of executions is very high cost.